src/js/containers/ImageOptimizationCard/ImageOptimizationCard.js

```diff
@@ -0,0 +1,63 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { FormattedMessage, injectIntl } from 'react-intl';
+import _ from 'lodash';
+import { asyncPluginUpdateSetting } from '../../actions/pluginSettings';
+import { getPluginSettingsValueForZoneId } from '../../selectors/pluginSettings';
+import { Card, CardSection, CardContent, CardDrawers } from 'cf-component-card';
+import CustomCardControl from '../../components/CustomCardControl/CustomCardControl';
+import { asyncZoneUpdateSetting } from '../../actions/zoneSettings';
+import Toggle from 'cf-component-toggle';
+
+const SETTING_NAME_MIRAGE = "mirage";
+const SETTING_NAME_POLISH = "polish";
+const MINIMUM_PLAN = "Pro Plan";
+
+class ImageOptimizationCard extends Component {
+
+    handleChange(value) {
+        let { activeZoneId, dispatch } = this.props;
+        value = (value === true ? "on" : "off");
+        dispatch(asyncZoneUpdateSetting(SETTING_NAME_MIRAGE, activeZoneId, value));
+        dispatch(asyncZoneUpdateSetting(SETTING_NAME_POLISH, activeZoneId, value));
+    }
+
+    render() {
+    	let { activeZone, config, zones } = this.props;
+    	let zone;
+        if(!_.isEmpty(activeZone.name)) {
+            zone = zones[activeZone.name];
+        }
+
+        const { formatMessage } = this.props.intl;
+        return (
+            <div>
+                <Card>
+                    <CardSection>
+                        <CardContent title={formatMessage({id: 'container.imageOptimization.title'})}>
+                            <p><FormattedMessage id="container.imageOptimization.description" /></p>
+                        </CardContent>
+                        <CustomCardControl minimumPlan={ MINIMUM_PLAN } currentPlan={ zone.plan.name }>
+                            <Toggle
+                                label=""
+                                value={(this.props.imageOptimizationValue == "on")}
+                                onChange={this.handleChange.bind(this)}/>
+                        </CustomCardControl>
+                    </CardSection>
+                </Card>
+            </div>
+        );
+    }
+}
+
+function mapStateToProps(state) {
+	var imageOptimizationValue = state.zoneSettings.entities[state.activeZone.id][SETTING_NAME_MIRAGE].value && 
```

Comment:
We want to avoid logic in `mapStateToProps(state)`.  Build a selectors/zoneSetttings.js selector to get this state if it doesn't already exist and then have one state value for mirage, and another for polish and do the `imageOptimizationValue` logic in `render()`.